File: studentmanagement-frontend/src/service/StudentService.js
import axios from "axios";

const BASE_URL = "http://54.169.167.73:8080/student";
class StudentService{

    //**Method to get all student from our api or database */
    getAllStudent(){
        return axios.get(BASE_URL);
    }
    /**MEthod to save student */
    saveStudent(studentData){
        return axios.post(BASE_URL, studentData);
    }
    updateStudent(id, studentData){
        return axios.put(`${BASE_URL}/${id}`, studentData)
    }
    getStudentById(id){
        return axios.get(`${BASE_URL}/${id}`);
    }
    deleteStudent(id){
        return axios.delete(BASE_URL +"/" +id);
    }

}
export default new StudentService();

File: studentmanagmentbackend/src/main/java/com/tutorial/studentmanagmentbackend/service/StudentService.java
package com.tutorial.studentmanagmentbackend.service;

import com.tutorial.studentmanagmentbackend.model.Student;
import com.tutorial.studentmanagmentbackend.repository.StudentRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Sort;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class StudentService implements StudentServiceInterface {

    @Autowired
    private StudentRepository studentRepository;

    @Override
    public Student saveStudent(Student student) {
        return studentRepository.save(student);
    }

    @Override
    public Optional<Student> getStudentById(int id) {
        return studentRepository.findById(id);
    }

    @Override
    public List<Student> getAllStudent() {
        return studentRepository.findAll(Sort.by(Sort.Direction.DESC, "id"));
    }

    @Override
    public Student updateStudent(int id, Student student) {
        Student studentToUpdate = studentRepository.findById(id).orElseThrow();
        studentToUpdate.setFirstName(student.getFirstName());
        studentToUpdate.setLastName(student.getLastName());
        studentToUpdate.setNim(student.getNim());
        studentToUpdate.setKelas(student.getKelas());
        studentToUpdate.setEmail(student.getEmail());
        return studentRepository.save(studentToUpdate);
    }

    @Override
    public void deleteStudent(int id) {
        studentRepository.deleteById(id);
    }
}
